### PR TITLE
EC2 IGW Calc Policy Fix

### LINKED ISSUE #393: Calculated policy - Fix aws_ec2_public_subnet calc policy
Calc policy was not general. If route table was main or had multiple associations, the associations request only got the first object in the array. the new query is made to get all associations, set a limit to 5000, and modifying the resulting nunjucks template.

### PR DESCRIPTION
**Pull request by choxie**
Calc policy was not general. If route table was main or had multiple associations, the associations request only got the first object in the array. the new query is made to get all associations, set a limit to 5000, and modifying the resulting nunjucks template.

Closes #393

### CODE REVIEW

File: calculated_policies/aws_ec2_public_subnet/main.tf [hunk 1]
```diff
@@ -19,35 +19,38 @@ resource "turbot_policy_setting" "instance_subnet_" {
...
   {%- set hasIGW = false -%}
   {%- for item in $.resources.items -%}
-    {%- if item.associations == $.resource.subnetId -%}
-      {%- for gateway in item.routes -%}
-        {%- if 'igw' in gateway.GatewayId -%}
-          {%- if hasIGW == false -%}
-            "Not approved"
-            {%- set hasIGW = true -%}
-          {%- endif -%}
+      {% for subnetid in item.associations %}
```
Comment (omerosaienni) [inline calculated_policies/aws_ec2_public_subnet/main.tf:39]: Should be:

for association in associations. 

Can you fix things like that on this PR and then ping me and we can merge it immediately

File: calculated_policies/aws_ec2_public_subnet/main.tf [hunk 2]
```diff
@@ -19,35 +19,38 @@ resource "turbot_policy_setting" "instance_subnet_" {
...
+        {%- if subnetid.SubnetId == $.resource.subnetId -%}
+            {%- for gateway in item.routes -%}
```
Comment (omerosaienni) [inline calculated_policies/aws_ec2_public_subnet/main.tf:41]: for route in routes

File: calculated_policies/aws_ec2_public_subnet/README.md [hunk 3]
```diff
@@ -42,21 +42,24 @@ When the Route Table entry is found then check each Route to find the Route with
...
 {%- set hasIGW = false -%}
 {%- for item in $.resources.items -%}
-  {%- if item.associations == $.resource.subnetId -%}
-    {%- for gateway in item.routes -%}
-      {%- if 'igw' in gateway.GatewayId -%}
-        {%- if hasIGW == false -%}
-          "Not approved"
-          {%- set hasIGW = true -%}
-        {%- endif -%}
+    {% for subnetid in item.associations %}
```
Comment (omerosaienni) [inline calculated_policies/aws_ec2_public_subnet/README.md:48]: See other comments

File: calculated_policies/aws_ec2_cross_account_ami_snapshot/main.tf [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  value           = "Check: Approved"
+}
+
+# Sets a calculated policy AWS > EC2 > AMI > Approved > Usage
+# Checks the snapshot shared account permissions and cross references them against the Turbot File.
+# If there is any account that does not exist in the Turbot File, this policy is set to "Not Approved."
+resource "turbot_policy_setting" "ec2_ami_approved_usage" {
+  resource        = turbot_smart_folder.ec2_trusted_accounts.id
+  type            = "tmod:@turbot/aws-ec2#/policy/types/amiApprovedUsage"
+  # GraphQL to pull policy Statements
```
Comment (omerosaienni) [inline calculated_policies/aws_ec2_cross_account_ami_snapshot/main.tf:83]: remove comment

File: calculated_policies/aws_ec2_cross_account_ami_snapshot/main.tf [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+# MODIFY THESE LISTS BEFORE APPLYING THE TERRAFORM PLAN!
+resource "turbot_file" "trusted_accounts_list" {
+  parent      = "tmod:@turbot/turbot#/"
+  title       = var.turbot_file_name
+  description = var.turbot_file_description
+  akas        = var.file_aka
+  content     = <<EOF
+  {
+    "snapshot_trusted_accounts" : [
+      "567890123456",
```
Comment (omerosaienni) [inline calculated_policies/aws_ec2_cross_account_ami_snapshot/main.tf:18]: make these non-generic:

11112222333 for example, should fail

**Review by omerosaienni — CHANGES_REQUESTED** [review]

File: calculated_policies/aws_ec2_public_subnet/main.tf (hunk 1 above)
Comment (choxie) [reply calculated_policies/aws_ec2_public_subnet/main.tf:39]: where does the associations come from? dont i need to use the item.associations format?

File: calculated_policies/aws_ec2_public_subnet/main.tf (hunk 2 above)
Comment (choxie) [reply calculated_policies/aws_ec2_public_subnet/main.tf:41]: same thing as above, dont i need to parse out the field with the items.routes format?

**Pull request closed without merging**